frontend/src/metabase/visualizations/components/FunnelNormal.jsx

```diff
@@ -26,7 +26,14 @@ export default class FunnelNormal extends Component {
     const dimensionIndex = 0;
     const metricIndex = 1;
     const cols = series[0].data.cols;
-    const rows = series.map(s => s.data.rows[0]);
+    const rows = settings["funnel.rows"]
+      ? settings["funnel.rows"]
+          .filter(fr => fr.enabled)
+          .map(
+            fr =>
+              series.find(s => s.card.rowIndex === fr.rowIndex).data.rows[0],
+          )
```

Comment:
I wonder if this is equivalent to
```suggestion
          .map(fr => series[fr.rowIndex].data.rows[0])
```
As it seems, `series` will keep the same order so we should be able to just access its index using `fr.rowIndex` directly, but I'm not sure if there will be a case that this will fail.

Reply:
appears to work!